### DISCUSSION TITLE
Why sbt new is Sooooo slow ! (for first time user)

### DISCUSSION THREAD

**Discussion by yinshurman** [scala/scala3#14979] [General Discussion]
Although already had heard that sbt is amazingly slow and anti-human , it still makes me shocked that the following simple commands for starting a simple project downloads so many irrelated packages and spent a couple of hours to finish.
` sbt new scala/scala3.g8 `
and after that 
`sbt run`
cost another couple of hours to fetch tons of packages and stuck in 
`root / update`
😅
Is this abnormal phenomenon normal in scala world?

sbt version 1.6.2
macos catalina 10.15.7

**Comment by bishabosha** [comment]
This seems surprising to me, and I'm sorry it has been so frustrating, may I ask what is your internet speed?

**Reply by yinshurman** [reply]
> This seems surprising to me, and I'm sorry it has been so frustrating, may I ask what is your internet speed?

my network bandwidth is about 90Mbps. 
and before  run `sbt  new scala/scala3.g8` , 
i just installed sbt by `sdk install sbt`, and there is jdk8 in my computer already configured, except those , nothing else about scala installed on my computer.
maybe this is the reason that  sbt's first run will download so many irrelated packages even the project is just a hello world?

**Reply by bishabosha** [reply]
I do remember sbt itself requires a lot of packages on its first run if you have not used it before, (but they are installed globally) so perhaps this can be intimidating to a new user who wants to write a hello world program

**Reply by bishabosha** [reply]
when you try to do `sbt new scala/scala3.g8` again in another directory is it still slow?

**Reply by yinshurman** [reply]
> when you try to do `sbt new scala/scala3.g8` again in another directory is it still slow?

YES ，It is !  
and i noticed that it is still download some packages i never heard , but i noticed the download speed is very slow,just about 5-10kib/s , from the default https/repo1.maven.org ,maybe this is the main reason , any way to change the default maven repository ?

**Comment by som-snytt** [comment]
Thanks, "irrelated" is now my word of the day. I guess the related word is "irritation".

I am amazed that modern sbt is so good, so it's a red flag that there is a rumor or pre-information that it is "anti-human". This is all in the context of how builds for the JVM work.

**Comment by nicolasstucki** [comment]
@yinshurman do you still have by any chance the terminal output from when this happened?

**Reply by yinshurman** [reply]
> @yinshurman do you still have by any chance the terminal output from when this happened?

I can't find that terminal output , so I just rerun `sbt new scala/scala3.g8`,this time it cost little time to finish this command ,but then I continued to run `sbt run `, it is still very slow and as I replied before, I noticed the download speed is very very slow ,just 5-15kib/s , but my network bandwidth is 90Mbps, and download other files is normally fast, so it seems the default repository ` https/repo1.maven.org` response slowly for my network?
